### PR TITLE
[Tabs] Bottom navigation support

### PR DESCRIPTION
**Pull request by brianjmoore**
This PR adds bottom navigation support to MDCTabBar by extending the UIBarPositioningDelegate protocol. Delegates which implement `positionForBar:` and return UIBarPositionBottom will cause the tab bar to be rendered as a bottom navigation bar.

### CODE REVIEW

File: components/Tabs/src/MDCTabBar.h [hunk 1]
```diff
@@ -33,9 +33,13 @@
...
-@interface MDCTabBar : UIView
+@interface MDCTabBar : UIView <UIBarPositioning>
 
-/** Return the desired height for the tab bar given an item appearance. */
+/** The default height for the tab bar given a position and item appearance. */
++ (CGFloat)defaultHeightForPosition:(UIBarPosition)position
+                     itemAppearance:(MDCTabBarItemAppearance)appearance;
+
+/** The default height for the tab bar in the top position, given an item appearance. */
 + (CGFloat)defaultHeightForItemAppearance:(MDCTabBarItemAppearance)appearance;
```
Comment (ianegordon) [inline components/Tabs/src/MDCTabBar.h]: Can we make this an instance method and use self.barPosition?
I'm a big fan of class methods when we don't need state, but in this case it seems like we need it.

File: components/Tabs/src/MDCTabBar.h [hunk 2]
```diff
@@ -133,9 +140,9 @@ IB_DESIGNABLE
 
 /**
  Delegate protocol for MDCTabBar. Clients may implement this protocol to receive notifications of
- selection changes in the tab bar.
+ selection changes in the tab bar or to change the bar's position-based appearance.
```
Comment (ianegordon) [inline components/Tabs/src/MDCTabBar.h]: Consider:
or return the bar's position.
(There is only one method in UIBarPositioningDelegate)

File: components/Tabs/src/MDCTabBar.m [hunk 3]
```diff
@@ -83,9 +73,23 @@ @interface MDCTabBar () <MDCItemBarDelegate>
 @implementation MDCTabBar {
   /// Item bar responsible for displaying the actual tab bar content.
   MDCItemBar *_itemBar;
+
+  // Flags tracking if properties are unset and using default values.
```
Comment (ianegordon) [inline components/Tabs/src/MDCTabBar.m:77]: I think we can implement this using UIAppearance proxy.
Can we catch up offline and discuss?

**Review by ianegordon — CHANGES_REQUESTED** [review]

File: components/Tabs/src/MDCTabBar.m [hunk 4]
```diff
@@ -39,24 +39,14 @@
...
 
-static MDCTabBarAlignment MDCTabBarAlignmentForItemBarAlignment(MDCItemBarAlignment alignment) {
-  switch (alignment) {
-    case MDCItemBarAlignmentCenter:
-      return MDCTabBarAlignmentCenter;
-
-    case MDCItemBarAlignmentLeading:
-      return MDCTabBarAlignmentLeading;
+/// Height for bottom navigation bars, in points.
+const CGFloat kBottomNavigationBarHeight = 56;
```
Comment (ianegordon) [inline components/Tabs/src/MDCTabBar.m]: nit: Add static.

File: components/Tabs/src/MDCTabBar.m [hunk 5]
```diff
@@ -39,24 +39,14 @@
...
 
-    case MDCItemBarAlignmentJustified:
-      return MDCTabBarAlignmentJustified;
+/// Maximum width for individual items in bottom navigation bars, in points.
+const CGFloat kBottomNavigationMaximumItemWidth = 168;
```
Comment (ianegordon) [inline components/Tabs/src/MDCTabBar.m]: nit: Add static.

File: components/Tabs/src/MDCTabBar.m [hunk 6]
```diff
@@ -39,24 +39,14 @@
...
 
-    case MDCItemBarAlignmentCenterSelected:
-      return MDCTabBarAlignmentCenterSelected;
-  }
-
-  NSCAssert(0, @"Invalid alignment value %zd", alignment);
-  return MDCTabBarAlignmentLeading;
-}
+/// Title-image padding for bottom navigation bars, in points.
+const CGFloat kBottomNavigationTitleImagePadding = 3;
```
Comment (ianegordon) [inline components/Tabs/src/MDCTabBar.m]: nit: Add static.

**Review by ianegordon — CHANGES_REQUESTED** [review]

**Review by ianegordon — CHANGES_REQUESTED** [review]
Add IB_Designable to NON-UI_APPEARANCE properties.

**Comment by brianjmoore** [comment]
Added IBInspectable to displaysUppercaseByDefault. The alignment and item appearance unfortunately are unsupported due to being enums so I opted to leave IBInspectable off to reduce confusion.

File: components/Tabs/src/MDCTabBar.m (hunk 6 above)
Comment (brianjmoore) [reply components/Tabs/src/MDCTabBar.m]: Done

File: components/Tabs/src/MDCTabBar.m (hunk 5 above)
Comment (brianjmoore) [reply components/Tabs/src/MDCTabBar.m]: Done

File: components/Tabs/src/MDCTabBar.m (hunk 4 above)
Comment (brianjmoore) [reply components/Tabs/src/MDCTabBar.m]: Done

File: components/Tabs/src/MDCTabBar.m (hunk 3 above)
Comment (brianjmoore) [reply components/Tabs/src/MDCTabBar.m:77]: Acknowledged. Discussed yesterday and determined that UIAppearance wouldn't be the right solution in this case.

File: components/Tabs/src/MDCTabBar.h (hunk 2 above)
Comment (brianjmoore) [reply components/Tabs/src/MDCTabBar.h]: Done

File: components/Tabs/src/MDCTabBar.h (hunk 1 above)
Comment (brianjmoore) [reply components/Tabs/src/MDCTabBar.h]: Made private for now

**Review by ianegordon — APPROVED** [review]
Thank you for your patience!
LGTM.

**Comment by ianegordon** [comment]
Tested Locally.

**Pull request merged**